core/src/main/java/org/hisp/dhis/android/core/user/loginconfig/LoginConfigObjectRepository.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright (c) 2004-2025, University of Oslo
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  Redistributions of source code must retain the above copyright notice, this
+ *  list of conditions and the following disclaimer.
+ *
+ *  Redistributions in binary form must reproduce the above copyright notice,
+ *  this list of conditions and the following disclaimer in the documentation
+ *  and/or other materials provided with the distribution.
+ *  Neither the name of the HISP project nor the names of its contributors may
+ *  be used to endorse or promote products derived from this software without
+ *  specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.android.core.user.loginconfig
+
+import io.reactivex.Single
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.rx2.rxSingle
+import org.hisp.dhis.android.core.arch.repositories.`object`.ReadOnlyObjectRepository
+
+class LoginConfigObjectRepository internal constructor(
+    private val loginConfigCall: LoginConfigCall,
+    private val serverUrl: String,
+) : ReadOnlyObjectRepository<LoginConfig> {
+    override fun get(): Single<LoginConfig?> {
+        return rxSingle { loginConfigCall.download(serverUrl) }.map { it }
+    }
+
+    override fun blockingGet(): LoginConfig {
+        return runBlocking { loginConfigCall.download(serverUrl) }
```

Comment:
I have tried running this method against a server lower than v41 and it throws this error. I don't think it is the error we would expect... Maybe we could throw a different error to avoid doing two requests (the first one to blockingExists and the second one to blockingGet). We could think about this when we define the login workflow
![image](https://github.com/user-attachments/assets/9c71e63c-e885-4952-82cd-c493189a771a)